axum-core/src/extract/tuple.rs

```diff
@@ -53,10 +48,10 @@ macro_rules! impl_from_request {
         // implementation of `FromRequest<S, B, Mut>` for `T: FromRequestParts<S>`.
         #[async_trait]
         #[allow(non_snake_case, unused_mut, unused_variables)]
-        impl<S, B, $($ty,)* $last> FromRequest<S, B, private::TupleOnce> for ($($ty,)* $last,)
+        impl<S, B, $($ty,)* $last> FromRequest<S, B> for ($($ty,)* $last,)
         where
             $( $ty: FromRequestParts<S> + Send, )*
-            $last: FromRequest<S, B, super::private::Once> + Send,
+            $last: FromRequest<S, B> + Send,
```

Comment:
If we want to write some docs about how to write generic extractors that wrap other extractors (that might or might not consume the request), this seems like pretty straight-forward advice: Implement both `FromRequestParts` and `FromRequest`, without any generic-ness over `M` in the latter case.

Reply:
Yes! Its on the todo list [here](https://github.com/tokio-rs/axum/pull/1272)